helix-view-aggregator/src/main/java/org/apache/helix/view/aggregator/HelixViewAggregator.java

```diff
@@ -191,14 +191,19 @@ private void handleSourceClusterEvent(ClusterViewEvent event) {
         _refreshViewCluster.set(true);
         break;
       case PeriodicViewRefresh:
-        if (!_refreshViewCluster.get()) {
+        // refresh local view cluster data cache
+        boolean cacheChanged = _viewClusterRefresher.refreshViewClusterDataCache();
+        if (cacheChanged) {
+          logger.info("Detected change in view cluster data, trigger a refresh");
+        } else if (!_refreshViewCluster.get()) {
```

Comment:
is refreshViewCluster ever getting set to FALSE ??

Reply:
Yes, https://github.com/apache/helix/blob/5a4cfc71a5bca3e5d87bafc627eb3b571deb162d/helix-view-aggregator/src/main/java/org/apache/helix/view/aggregator/HelixViewAggregator.java#L355